kolibri/plugins/facility/assets/src/views/ManageClassPage/index.vue

```diff
@@ -178,10 +340,34 @@
           classroom,
         ]);
       },
+      dropDownOptions() {
+        return [
+          {
+            label: this.copyClasslabel$(),
+            value: 'COPY_CLASS',
+            id: 'copy',
+          },
+          {
+            label: this.renameClassLabel$(),
+            value: 'EDIT_CLASS_NAME',
+            id: 'rename',
+          },
+          {
+            label: this.$tr('deleteClass'),
+            value: 'DELETE_CLASS',
+            id: 'delete',
+          },
+        ];
+      },
+      isClassNameInvalid() {
+        return this.tableRows.some(row => row[0] === this.copiedClassName);
+      },
```

Comment:
This is flashing the message to the user before closing the side panel when the copying is successful. 

This condition will be evaluated before the completion of `handleSubmitClassCopy` because once this line is evaluated:

`this.$store.commit('classManagement/SET_STATE', { classes: updatedClasses });`

That's where the `tableRows` is updated. You could maybe try closing the side panel before calling that `commit` in the function to mitigate this issue.

Reply:
@AllanOXDi I'm still seeing this flash.

Perhaps a flag like `this.submitting` or `this.loading` could be added and set to true until the very end of the `handleSubmitClassCopy` function where you set it back.

Then in the `isClassNameInvalid` add a `&& !this.submitting` to the end of it.